source/build.py

```diff
@@ -0,0 +1,168 @@
+        if '/ja/' in self.targetFullPath:
+            self.pruneLangs.add('en')
+
+    def handle_decl(self, data):
+        if data == "DOCTYPE html":
+            self.targetIsFullHTML = True
+
+            if not ('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath):
+                self.pruneLangs.add('en')
+                self.pruneLangs.add('ja')
+
+    def handle_starttag(self, tag, attrs):
+        position = self.getpos()[0]
+        if self.targetIsFullHTML and tag == 'body':
+            # mark for injection of Google Tag Manager Script.
+            self.markers.add(('injectTagManager', position))
+
+        if ( tag == 'a' and (('href', '..') in attrs) and
+                            (('class', '') in attrs)
+           ):
+                # mark for removal: unelegant blank TOC item.
+                self.markers.add(('prune', position))
+
+        for attr in attrs:
+            if isPruneLangElement(self, tag, attr):
+                # mark for removal: TOC items of the other language.
+                self.markers.add(('prune', position))
+
+                # retroactively prevent removal of main links from
+                #  top level page to each language's index page.
+                if isExceptionTopIndex(self, attr):
+                    self.markers.remove(('prune', position))
+
+def isPruneLangElement(self, tag, attr):
+    return ((
+        'ja' in self.pruneLangs and tag == 'a' and attr[0] == 'href' and
+        ('../ja/' in attr[1] or attr[1].startswith('ja/'))
+    ) or (
+        'en' in self.pruneLangs and tag == 'a' and attr[0] == 'href' and
+        ('../en/' in attr[1] or attr[1].startswith('en/'))
+    ))
+
+def isExceptionTopIndex(self, attr):
+    return (
+        not('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath) and
+        self.targetFullPath.endswith('/index.html') and
+        (attr[1] == 'en/' or attr[1] == 'ja/')
+    )
+
+# ============================================================================
+
+
+if ('mkdocs.yml' in lsResults) and ('docs' in lsResults) and isMkdocsInstalled:
+    print('All dependencies present. Initiate build...')
+    subprocess.call('mkdocs build --clean', shell=True)
+    print('Mkdocs build successful.')
+    siteDirectory = cwd + "/site"
+    htmlFileFullPaths = set([])
+
+    print('Crawling html files...')
+    for root, subDirectories, files in os.walk(siteDirectory):
+        for filename in os.listdir(root):
+            if filename.endswith('.html'):
+                htmlFileFullPaths.add(root + '/' + filename)
+                logging.debug(root + '/' + filename + ' found!')
+
+    for htmlFileFullPath in htmlFileFullPaths:
+        logging.debug('Parsing and marking ' + htmlFileFullPath)
+        readHtmlFile = open(htmlFileFullPath, 'r')
+        htmlString = readHtmlFile.read()
+        parser = PostMkdocsParser(htmlFileFullPath)
+        parser.feed(htmlString)
+
+        readHtmlFile.close()
```

Comment:
Place this line right after the `htmlString = readHtmlFile.read()` would be nice.
or don't use htmlString and just directly give `readHtmlFile.read()` into `parser.feed` function.